monkey/infection_monkey/i_puppet/target_host.py

```diff
@@ -22,6 +24,9 @@ class TargetHost(MutableInfectionMonkeyBaseModel):
     icmp: bool = Field(default=False)
     ports_status: TargetHostPorts = Field(default=TargetHostPorts())
 
+    def filter_tcp_ports(self, filter: FilterPortFunc) -> Sequence[PortScanData]:
+        return [p for p, d in self.ports_status.tcp_ports.items() if filter(d)]
+
```

Comment:
IDK that this should really be a function of `TargetHost`. In any case, this is somewhat less efficient than looking up the ports in the `tcp_ports` dict. You need to loop through all ports instead of just inspecting the ones you need (SMB ports, in this case).